Make generate is broken

```
go install  -ldflags '-extldflags "-static"' sigs.k8s.io/cluster-api-provider-aws/vendor/github.com/golang/mock/mockgen
can't load package: package sigs.k8s.io/cluster-api-provider-aws/vendor/github.com/golang/mock/mockgen: cannot find package "sigs.k8s.io/cluster-api-provider-aws/vendor/github.com/golang/mock/mockgen" in any of:
        /usr/lib/golang/src/sigs.k8s.io/cluster-api-provider-aws/vendor/github.com/golang/mock/mockgen (from $GOROOT)
        /home/alukiano/go/src/sigs.k8s.io/cluster-api-provider-aws/vendor/github.com/golang/mock/mockgen (from $GOPATH)
make: *** [Makefile:46: gencode] Error 1
```

Folder mockgen does not exist under the master of 
```
ll ~/go/src/sigs.k8s.io/cluster-api-provider-aws/vendor/github.com/golang/mock
total 24
Mar 17 14:01 AUTHORS
Mar 17 14:01 CONTRIBUTORS
Mar 17 14:01 gomock
Mar 17 14:01 LICENSE
```

And it does not good practice to reference tools from some external repository.

--- a/Gopkg.lock
+++ b/Gopkg.lock
@@ -98,7 +98,11 @@
 [[projects]]
   digest = "1:7912bb30698616cf0c41e5ef6a29a447c9a3e76bfcacbc21420ed3cdf616fd94"
   name = "github.com/golang/mock"
-  packages = ["gomock"]
+  packages = [
+    "gomock",
+    "mockgen",
+    "mockgen/model",
+  ]
   pruneopts = "T"
   revision = "51421b967af1f557f93a59e0057aaf15ca02e29c"
   version = "v1.2.0"
@@ -1040,6 +1044,7 @@
     "github.com/ghodss/yaml",
     "github.com/golang/glog",
     "github.com/golang/mock/gomock",
+    "github.com/golang/mock/mockgen",
     "github.com/libvirt/libvirt-go",
     "github.com/libvirt/libvirt-go-xml",
     "github.com/onsi/ginkgo",

--- a/Gopkg.toml
+++ b/Gopkg.toml
@@ -2,6 +2,7 @@ required = [
     "github.com/emicklei/go-restful",
     "github.com/onsi/ginkgo", # for test framework
     "github.com/onsi/gomega", # for test matchers
+    "github.com/golang/mock/mockgen", # for mock generation
     "k8s.io/client-go/plugin/pkg/client/auth/gcp", # for development against gcp
     "k8s.io/code-generator/cmd/deepcopy-gen", # for go generate
     "sigs.k8s.io/controller-tools/cmd/controller-gen", # for crd/rbac generation
@@ -25,6 +26,10 @@ required = [
 [[constraint]]
   name = "github.com/davecgh/go-spew"
   version = "1.1.1"
+
+[[constraint]]
+  name = "github.com/golang/mock"
+  version = "1.2.0"
 
 [[constraint]]
   name = "github.com/ghodss/yaml"

--- a/Makefile
+++ b/Makefile
@@ -43,7 +43,7 @@ generate: gendeepcopy gencode
 
 .PHONY: gencode
 gencode:
-	go install $(GOGCFLAGS) -ldflags '-extldflags "-static"' sigs.k8s.io/cluster-api-provider-aws/vendor/github.com/golang/mock/mockgen
+	go install $(GOGCFLAGS) -ldflags '-extldflags "-static"' github.com/openshift/cluster-api-provider-libvirt/vendor/github.com/golang/mock/mockgen
 	go generate ./pkg/... ./cmd/...
 
 .PHONY: gendeepcopy

--- a/pkg/cloud/libvirt/client/mock/client_generated.go
+++ b/pkg/cloud/libvirt/client/mock/client_generated.go
@@ -36,6 +36,7 @@ func (m *MockClient) EXPECT() *MockClientMockRecorder {
 
 // Close mocks base method
 func (m *MockClient) Close() error {
+	m.ctrl.T.Helper()
 	ret := m.ctrl.Call(m, "Close")
 	ret0, _ := ret[0].(error)
 	return ret0
@@ -43,11 +44,13 @@ func (m *MockClient) Close() error {
 
 // Close indicates an expected call of Close
 func (mr *MockClientMockRecorder) Close() *gomock.Call {
+	mr.mock.ctrl.T.Helper()
 	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "Close", reflect.TypeOf((*MockClient)(nil).Close))
 }
 
 // CreateDomain mocks base method
 func (m *MockClient) CreateDomain(arg0 client.CreateDomainInput) error {
+	m.ctrl.T.Helper()
 	ret := m.ctrl.Call(m, "CreateDomain", arg0)
 	ret0, _ := ret[0].(error)
 	return ret0
@@ -55,11 +58,13 @@ func (m *MockClient) CreateDomain(arg0 client.CreateDomainInput) error {
 
 // CreateDomain indicates an expected call of CreateDomain
 func (mr *MockClientMockRecorder) CreateDomain(arg0 interface{}) *gomock.Call {
+	mr.mock.ctrl.T.Helper()
 	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "CreateDomain", reflect.TypeOf((*MockClient)(nil).CreateDomain), arg0)
 }
 
 // DeleteDomain mocks base method
 func (m *MockClient) DeleteDomain(name string) error {
+	m.ctrl.T.Helper()
 	ret := m.ctrl.Call(m, "DeleteDomain", name)
 	ret0, _ := ret[0].(error)
 	return ret0
@@ -67,11 +72,13 @@ func (m *MockClient) DeleteDomain(name string) error {
 
 // DeleteDomain indicates an expected call of DeleteDomain
 func (mr *MockClientMockRecorder) DeleteDomain(name interface{}) *gomock.Call {
+	mr.mock.ctrl.T.Helper()
 	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "DeleteDomain", reflect.TypeOf((*MockClient)(nil).DeleteDomain), name)
 }
 
 // DomainExists mocks base method
 func (m *MockClient) DomainExists(name string) (bool, error) {
+	m.ctrl.T.Helper()
 	ret := m.ctrl.Call(m, "DomainExists", name)
 	ret0, _ := ret[0].(bool)
 	ret1, _ := ret[1].(error)
@@ -80,11 +87,13 @@ func (m *MockClient) DomainExists(name string) (bool, error) {
 
 // DomainExists indicates an expected call of DomainExists
 func (mr *MockClientMockRecorder) DomainExists(name interface{}) *gomock.Call {
+	mr.mock.ctrl.T.Helper()
 	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "DomainExists", reflect.TypeOf((*MockClient)(nil).DomainExists), name)
 }
 
 // LookupDomainByName mocks base method
 func (m *MockClient) LookupDomainByName(name string) (*libvirt_go.Domain, error) {
+	m.ctrl.T.Helper()
 	ret := m.ctrl.Call(m, "LookupDomainByName", name)
 	ret0, _ := ret[0].(*libvirt_go.Domain)
 	ret1, _ := ret[1].(error)
@@ -93,11 +102,13 @@ func (m *MockClient) LookupDomainByName(name string) (*libvirt_go.Domain, error)
 
 // LookupDomainByName indicates an expected call of LookupDomainByName
 func (mr *MockClientMockRecorder) LookupDomainByName(name interface{}) *gomock.Call {
+	mr.mock.ctrl.T.Helper()
 	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "LookupDomainByName", reflect.TypeOf((*MockClient)(nil).LookupDomainByName), name)
 }
 
 // CreateVolume mocks base method
 func (m *MockClient) CreateVolume(arg0 client.CreateVolumeInput) error {
+	m.ctrl.T.Helper()
 	ret := m.ctrl.Call(m, "CreateVolume", arg0)
 	ret0, _ := ret[0].(error)
 	return ret0
@@ -105,11 +116,13 @@ func (m *MockClient) CreateVolume(arg0 client.CreateVolumeInput) error {
 
 // CreateVolume indicates an expected call of CreateVolume
 func (mr *MockClientMockRecorder) CreateVolume(arg0 interface{}) *gomock.Call {
+	mr.mock.ctrl.T.Helper()
 	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "CreateVolume", reflect.TypeOf((*MockClient)(nil).CreateVolume), arg0)
 }
 
 // VolumeExists mocks base method
 func (m *MockClient) VolumeExists(name string) (bool, error) {
+	m.ctrl.T.Helper()
 	ret := m.ctrl.Call(m, "VolumeExists", name)
 	ret0, _ := ret[0].(bool)
 	ret1, _ := ret[1].(error)
@@ -118,11 +131,13 @@ func (m *MockClient) VolumeExists(name string) (bool, error) {
 
 // VolumeExists indicates an expected call of VolumeExists
 func (mr *MockClientMockRecorder) VolumeExists(name interface{}) *gomock.Call {
+	mr.mock.ctrl.T.Helper()
 	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "VolumeExists", reflect.TypeOf((*MockClient)(nil).VolumeExists), name)
 }
 
 // DeleteVolume mocks base method
 func (m *MockClient) DeleteVolume(name string) error {
+	m.ctrl.T.Helper()
 	ret := m.ctrl.Call(m, "DeleteVolume", name)
 	ret0, _ := ret[0].(error)
 	return ret0
@@ -130,5 +145,6 @@ func (m *MockClient) DeleteVolume(name string) error {
 
 // DeleteVolume indicates an expected call of DeleteVolume
 func (mr *MockClientMockRecorder) DeleteVolume(name interface{}) *gomock.Call {
+	mr.mock.ctrl.T.Helper()
 	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "DeleteVolume", reflect.TypeOf((*MockClient)(nil).DeleteVolume), name)
 }

